core/src/main/java/org/elasticsearch/index/query/GeoDistanceQueryParser.java

```diff
@@ -130,8 +130,13 @@ public GeoDistanceQueryBuilder fromXContent(QueryParseContext parseContext) thro
                 } else if ("validation_method".equals(currentFieldName)) {
                     validationMethod = GeoValidationMethod.fromString(parser.text());
                 } else {
-                    point.resetFromString(parser.text());
-                    fieldName = currentFieldName;
+                    if (fieldName == null) {
+                        point.resetFromString(parser.text());
+                        fieldName = currentFieldName;
+                    } else {
+                        throw new ParsingException(parser.getTokenLocation(), "[" + GeoDistanceQueryBuilder.NAME + "] point specified twice. [" + currentFieldName
```

Comment:
Sure, I can change the wording. However I think this can happen here only when the parser found more than one field in the query that the parser interprets as field name for the point, basically overwriting any previously parsed point. I might change to outputting the assigned current `fieldName` value and the `currentFieldName` so debugging the query is easier, wdyt?